### PR TITLE
automatic skip only for Apps without Skip Button

### PR DESCRIPTION
**Pull request by jumoog**
#206

### CODE REVIEW

**Comment by jumoog** [comment]
> Eh. The auto skip option is the only way to have this plugin work with [Jellyfin for Kodi](https://github.com/jellyfin/jellyfin-kodi).

whats the ClientName?

**Comment by AbandonedCart** [comment]
Why not future proof it? Android TV and Kodi can be your defaults and allow entry of user defined values.

**Comment by jumoog** [comment]
Good idea. I would go one step further and let the user choose the client from his known clients list.

**Comment by jumoog** [comment]
PoC:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Drag and Drop Multiple Items</title>
    <style>
        .container {
            display: flex;
            justify-content: center;
            align-items: center;
            gap: 20px;
            margin-top: 50px;
        }
        ul {
            list-style-type: none;
            padding: 0;
            width: 200px;
            height: 300px;
            border: 1px solid #ccc;
            overflow-y: auto;
        }
        li {
            padding: 10px;
            cursor: pointer;
            border-bottom: 1px solid #ccc;
            user-select: none;
        }
        li:hover {
            background-color: #f0f0f0;
        }
        li.selected {
            background-color: #d0e0ff;
        }
    </style>
</head>
<body>

<div class="container">
    <ul id="leftList">
        <li draggable="true">Jellyfin Web</li>
        <li draggable="true">Jellyfin Android</li>
        <li draggable="true">Findroid</li>
        <li draggable="true">Jellyfin Media Player</li>
    </ul>
    <ul id="rightList">
        <li draggable="true">Android TV</li>
        <li draggable="true">Kodi</li>
    </ul>
</div>

<script>
    document.querySelectorAll('ul').forEach(list => {
        list.addEventListener('click', function(event) {
            if (event.target.tagName === 'LI') {
                event.target.classList.toggle('selected');
            }
        });

        list.addEventListener('dragstart', function(event) {
            if (event.target.tagName === 'LI') {
                const selectedItems = list.querySelectorAll('li.selected');
                if (!selectedItems.length) {
                    event.target.classList.add('selected');
                }
                event.dataTransfer.setData('text/plain', null);
            }
        });

        list.addEventListener('dragover', function(event) {
            event.preventDefault();
        });

        list.addEventListener('drop', function(event) {
            event.preventDefault();
            const selectedItems = document.querySelectorAll('.selected');
            selectedItems.forEach(item => {
                list.appendChild(item);
                item.classList.remove('selected');
            });
        });
    });

    document.querySelectorAll('li').forEach(item => {
        item.setAttribute('draggable', 'true');
    });
</script>

</body>
</html>
```

**Comment by AbandonedCart** [comment]
Findroid uses the plugin API and displays a native button.

**Comment by dredstone1** [comment]
Add the option to auto skip when the player isn't currently active, for example when the user watch through pip.
Is that even possible to detect?

**Comment by AbandonedCart** [comment]
It depends on the situation. In a native app, pip is handled by the app. On a website, it's handled by the browser. The browser would expose that information to JavaScript. The app may not.

https://stackoverflow.com/questions/61507068/checking-whether-or-not-a-video-is-picture-in-picture-mode

**Comment by dredstone1** [comment]
The app version don't support pip, and from what you sent the web client can detect it, so it something that you can do?

**Comment by AbandonedCart** [comment]
> The app version don't support pip, and from what you sent the web client can detect it, so it something that you can do?

#208 is the PR for this feature.

**Comment by dredstone1** [comment]
OK I will add issue. And maybe in the future I will code it myself

**Comment by AbandonedCart** [comment]
> OK I will add issue. And maybe in the future I will code it myself

The point of that was actually to be patient.

File: ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs [hunk 1]
```diff
@@ -112,6 +112,12 @@ private void PlaybackTimer_Elapsed(object? sender, ElapsedEventArgs e)
     {
         foreach (var session in _sessionManager.Sessions)
         {
+            // only need for official Android TV App and jellyfin-kodi
+            if (session.Client != "Android TV" || session.Client != "Kodi")
```
Comment (rlauuzo) [inline ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs:116]: It’s probably a good idea to auto-skip if injecting fails with an 'UnableToAddSkipButton' error.

Comment (jumoog) [reply ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs:116]: Yes thats a good idea

Comment (AbandonedCart) [reply ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs]: There should also be a log printed that that is why it’s automatically skipping. You know that’ll be at least one post in the issues.

Comment (jumoog) [reply ConfusedPolarBear.Plugin.IntroSkipper/AutoSkip.cs]: i did

**Comment by jumoog** [comment]
f

> Why not future proof it? Android TV and Kodi can be your defaults and allow entry of user defined values. This also allows you to add new values in the future by simply updating a list.

will do this in another PR

**Comment by AbandonedCart** [comment]
The frontend will no doubt attract more posts about possible selections anyway, so keeping them separate will make it more manageable.

**Pull request merged**